pkg/chunk/aws_storage_client.go

```diff
@@ -477,7 +477,7 @@ func (a awsStorageClient) getDynamoDBChunks(ctx context.Context, chunks []Chunk)
 	result := []Chunk{}
 	unprocessed := dynamoDBReadRequest{}
 	backoff, numRetries := minBackoff, 0
-	for outstanding.Len()+unprocessed.Len() > 0 {
+	for outstanding.Len()+unprocessed.Len() > 0 && numRetries < maxRetries {
```

Comment:
There's no error thrown if we exceed maxRetries, previously the ingester would try forever, right?